usecase/parser/v0_42_7/ibcmsg/ibcmsg.go

```diff
@@ -0,0 +1,147 @@
+	var rawMsg ibc_model.RawMsgRecvPacket
+	decoderConfig := &mapstructure.DecoderConfig{
+		WeaklyTypedInput: true,
+		DecodeHook: mapstructure.ComposeDecodeHookFunc(
+			mapstructure.StringToTimeDurationHookFunc(),
+			mapstructure.StringToTimeHookFunc(time.RFC3339),
+			utils.StringToDurationHookFunc(),
+			utils.StringToByteSliceHookFunc(),
+		),
+		Result: &rawMsg,
+	}
+	decoder, decoderErr := mapstructure.NewDecoder(decoderConfig)
+	if decoderErr != nil {
+		panic(fmt.Errorf("error creating RawMsgRecvPacket decoder: %v", decoderErr))
+	}
+	if err := decoder.Decode(parserParams.Msg); err != nil {
+		panic(fmt.Errorf("error decoding RawMsgRecvPacket: %v", err))
+	}
+
+	if !ibcmsg.IsPacketMsgTransfer(rawMsg.Packet) {
+		// unsupported application
+		return []command.Command{}
+	}
+
+	// Transfer application, MsgTransfer
+	var rawFungibleTokenPacketData ibc_model.FungibleTokenPacketData
+	rawPacketData := base64_internal.MustDecodeString(rawMsg.Packet.Data)
+	json.MustUnmarshal(rawPacketData, &rawFungibleTokenPacketData)
+
+	if !parserParams.MsgCommonParams.TxSuccess {
+		msgRecvPacketParams := ibc_model.MsgRecvPacketParams{
+			RawMsgRecvPacket: rawMsg,
+
+			Application: "transfer",
+			MessageType: "MsgTransfer",
+			MaybeFungibleTokenPacketData: &ibc_model.MsgRecvPacketFungibleTokenPacketData{
+				FungibleTokenPacketData: rawFungibleTokenPacketData,
+			},
+		}
+
+		return []command.Command{command_usecase.NewCreateMsgIBCRecvPacket(
+			parserParams.MsgCommonParams,
+
+			msgRecvPacketParams,
+		)}
+	}
+
+	log := utils.NewParsedTxsResultLog(&parserParams.TxsResult.Log[parserParams.MsgIndex])
+
+	recvPacketEvent := log.GetEventByType("recv_packet")
+	if recvPacketEvent == nil {
+		panic("missing `recv_packet` event in TxsResult log")
+	}
+
+	fungibleTokenPacketEvent := log.GetEventByType("fungible_token_packet")
+	if fungibleTokenPacketEvent == nil {
+		// Note: this could happen when the packet is already relayed.
+		// https://github.com/cosmos/ibc-go/blob/760d15a3a55397678abe311b7f65203b2e8437d6/modules/core/04-channel/keeper/packet.go#L239
+		// https://github.com/cosmos/ibc-go/blob/760d15a3a55397678abe311b7f65203b2e8437d6/modules/core/keeper/msg_server.go#L508
+
+		packetAck := ibc_model.MsgRecvPacketPacketAck{
+			MaybeError: primptr.String("missing `fungible_token_packet` event in TxsResult log, this could happen when the packet is already relayed"),
+		}
+
+		msgRecvPacketParams := ibc_model.MsgRecvPacketParams{
+			RawMsgRecvPacket: rawMsg,
+
+			Application: "transfer",
+			MessageType: "MsgTransfer",
+			MaybeFungibleTokenPacketData: &ibc_model.MsgRecvPacketFungibleTokenPacketData{
+				FungibleTokenPacketData: rawFungibleTokenPacketData,
+				Success:                 false,
+			},
+
+			PacketSequence: typeconv.MustAtou64(recvPacketEvent.MustGetAttributeByKey("packet_sequence")),
+			// TODO: Remove this when IBCChannel projection no longer relies on packetAck.MaybeError to check if MsgRecvPacket is success or not
```

Comment:
Cc @ysong42 , after this PR, we should be able other up the projection right?

Reply:
Sorry, what do you mean by `other up`? 😅

Reply:
Just pulled the latest code and checked.

The projection related logic is not updated. After merging this PR, let me create another PR to fix those related TODOs.